pkg/daemon/daemon.go

```diff
@@ -105,23 +108,86 @@ func (d *Daemon) tick() {
 			continue
 		}
 		d.providerRequestsCounter.WithLabelValues(providerName, "success").Inc()
-		for _, project := range providerProjects {
-			err = d.cloneOrPull(project)
+		for _, proj := range providerProjects {
+			err = d.cloneOrPull(proj)
 			if err != nil {
-				d.logger.Error(fmt.Sprintf("fail to retrieve project %s: %s", project.Name, err.Error()))
+				d.logger.Error(fmt.Sprintf("fail to retrieve project %s: %s", proj.Name, err.Error()))
 				d.gitRequestsCounter.WithLabelValues("failure").Inc()
 				continue
 			}
 			d.gitRequestsCounter.WithLabelValues("success").Inc()
+
+			codeowners, err := codeowners.NewCodeowners(proj.Path)
+			if err != nil {
+				d.logger.Warn(fmt.Sprintf("Failed to parse CODEOWNERS for project %s: %s", proj.Name, err.Error()))
+			}
+			// Create subprojects based on expanded paths
+			err = d.exploreProjectFiles(proj.Path, codeowners, proj, &subProjects)
```

Comment:
How much time this function takes for large projects ? We don't have an easy way to explore only a subset of directories for example to speed it up ?

Reply:
Example execution with 4 projects and one group applied to files as well:
```
➜  standards-insights git:(test-subproject) ✗ standards-insights server --config config.example.yaml   
time=2024-08-19T10:15:54.491+02:00 level=INFO msg="version development (commit unknown) released unknown"
time=2024-08-19T10:15:54.492+02:00 level=INFO msg="starting HTTP server on 127.0.0.1:3002"
time=2024-08-19T10:15:54.492+02:00 level=INFO msg="starting daemon"
time=2024-08-19T10:15:54.492+02:00 level=INFO msg="checking projects"
time=2024-08-19T10:16:36.997+02:00 level=WARN msg="Failed to parse CODEOWNERS for project 1: no CODEOWNERS found in CODEOWNERS"
time=2024-08-19T10:16:39.442+02:00 level=WARN msg="Failed to parse CODEOWNERS for project 2: no CODEOWNERS found in CODEOWNERS"
time=2024-08-19T10:16:39.500+02:00 level=INFO msg="checking project 3"
time=2024-08-19T10:16:45.028+02:00 level=INFO msg="checking project 1"
time=2024-08-19T10:16:46.669+02:00 level=INFO msg="checking project 4"
time=2024-08-19T10:16:47.846+02:00 level=INFO msg="checking project 2"
time=2024-08-19T10:16:49.078+02:00 level=INFO msg="projects checked"
```
Project 3 and 4 are huge projects including, especially one is including 11165 files. So I would say that it's a reasonable amount of time considering how the metrics are consumed. Also, keep in mind that later on we're applying a `files-pattern` to limit the number of files the group checks are applied to.